packages/@aws-cdk/aws-stepfunctions-tasks/lib/sns/publish.ts

```diff
@@ -77,6 +133,17 @@ export class SnsPublish extends sfn.TaskStateBase {
       }
     }
 
+    if (props.messageAttributes) {
+      const attributes: { [key: string]: any } = {};
+      for (const attrib of props.messageAttributes) {
+        const dataType = attrib.type || SnsMessageAttributeType.STRING;
+        attributes[attrib.name] = {
+          DataType: dataType,
+          StringValue: attrib.value.value,
+        };
+      };
```

Comment:
nit: would be nice to move this into a utility method to keep the constructor minimal.